chromium/background.js

```diff
@@ -88,26 +92,35 @@ loadStoredUserRules();
 
 /**
  * Set the icon color correctly
- * Depending on http-nowhere it should be red/default
+ * inactive: extension is enabled, but no rules were triggered on this page.
+ * blocking: extension is in "block all HTTP requests" mode.
+ * active: extension is enabled and rewrote URLs on this page.
+ * disabled: extension is disabled from the popup menu.
  */
-var setIconColor = function() {
-  var newIconPath = httpNowhereOn ? './icon38-red.png' : './icon38.png';
-  chrome.browserAction.setIcon({
-    path: newIconPath
+var updateState = function() {
+  chrome.tabs.query({active: true}, function(tabs) {
```

Comment:
Fixed, thanks.

This has the unfortunate side effect that if you have two windows visible at once, the icon in both windows will reflect the state of whichever window is currently focused, and switching windows will change icon state (if it's different). We could probably fix this by switching back from browserAction to pageAction, but I'm not very excited about that. Probably the current behavior is fine.